packages/vaadin-grid/src/vaadin-grid-selection-column.js

```diff
@@ -197,15 +120,84 @@ class GridSelectionColumnElement extends GridColumnElement {
   connectedCallback() {
     super.connectedCallback();
     if (this._grid) {
-      this._grid.addEventListener('active-item-changed', this._boundOnActiveItemChanged);
-      this._grid.addEventListener('data-provider-changed', this._boundOnDataProviderChanged);
-      this._grid.addEventListener('filter-changed', this._boundOnSelectedItemsChanged);
-      this._grid.addEventListener('selected-items-changed', this._boundOnSelectedItemsChanged);
+      this._grid.addEventListener('active-item-changed', this.__boundOnActiveItemChanged);
+      this._grid.addEventListener('data-provider-changed', this.__boundOnDataProviderChanged);
+      this._grid.addEventListener('filter-changed', this.__boundOnSelectedItemsChanged);
+      this._grid.addEventListener('selected-items-changed', this.__boundOnSelectedItemsChanged);
+    }
+  }
+
+  /**
+   * Overrides the path header renderer to render the Select All checkbox
+   * in the header cell even the path property has been defined.
+   *
+   * @private
+   */
+  __pathHeaderRenderer(...args) {
+    return this.__defaultHeaderRenderer(...args);
```

Comment:
@tomivirkki Previously, it worked the same way, it used to ignore path property for the header cell. 

Though, I have some doubts about whether this logic is correct. I would like to have your approval.